lib/screens/home/products/widgets/product_item.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:flutter/material.dart';
+import 'package:shop_app/models/product/models.dart';
+
+class ProductItem extends StatelessWidget {
+  const ProductItem({
+    super.key,
+    required this.product,
+  });
+
+  final Product product;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      padding: const EdgeInsets.all(5),
+      decoration: ShapeDecoration(
+        shape: RoundedRectangleBorder(
+          borderRadius: BorderRadius.circular(12),
+          side: const BorderSide(
+            color: Colors.grey,
+          ),
+        ),
+        color: Colors.white,
+      ),
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: [
+          Expanded(
+            child: SizedBox.square(
+              dimension: MediaQuery.of(context).size.height / 4,
```

Comment:
If you are using MediaQuery it is better to save needed value in variable before return type.

Also for better app perfomance use MediaQuery.sizeOf(context).height;